src/GafferImage/Merge.cpp

```diff
@@ -308,7 +308,9 @@ IECore::ConstFloatVectorDataPtr Merge::merge( F f, const std::string &channelNam
 		ConstFloatVectorDataPtr channelData;
 		ConstFloatVectorDataPtr alphaData;
 
-		if( ImageAlgo::channelExists( channelNames, channelName ) )
+		const Box2i validBound = boxIntersection( tileBound, dataWindow );
+
+		if( ImageAlgo::channelExists( channelNames, channelName ) && !BufferAlgo::empty( validBound ) )
```

Comment:
We need to update the matching tests in `hashChannelData()` so that they continue to match. Since we seem to be relying on the OpenImageIOReader to detect these problems, do you think it's worth updating `OpenImageIOReader::hashChannelData()` to throw the same error that `OpenImageIO::computeChannelData()` is throwing?